src/components/Tabs/Tabs.tsx

```diff
@@ -65,66 +61,37 @@ export const Tabs = ({
   onChange,
   ...other
 }: Props) => {
-  const ref = useRef<number | undefined>();
+  const getUID = useUIDSeed();
+  const activeTabPanelId = useUID();
   const headerRef = useRef<HTMLDivElement>(null);
   const [activeIndexState, setActiveIndexState] = useState(activeIndex);
   const [scrollableLeft, setScrollableLeft] = useState<boolean>(false);
   const [scrollableRight, setScrollableRight] = useState<boolean>(false);
 
-  /** Get all children that are Tab components. Ignore any others. */
-  const getTabs = useCallback(() => {
+  /** Children that are actually tabs. Any others are ignored. */
+  const tabs = useMemo(() => {
     return allByType(children, Tab);
   }, [children]);
 
-  const tabRefs = getTabs().map(() => React.createRef<HTMLAnchorElement>());
-  const [tabPanelIds, setTabPanelIds] = useState<string[]>([]);
-  const [tabIds, setTabIds] = useState<string[]>([]);
-  const getUID = useUIDSeed();
+  const tabRefs = useMemo(
+    () => tabs.map(() => React.createRef<HTMLAnchorElement>()),
+    [tabs],
+  );
```

Comment:
We only need to re-generate refs when tabs themselves are regenerated, instead of on every render.

Reply:
I think this is pretty similar to the `useRef([])` solution [discussed at our team meeting](https://docs.google.com/document/d/1xnmVlJALAWiSNZdgr3aWWiki0GY2rEaDu_UDEKktocc/edit#), except better/easier/simpler.